src/EditorFeatures/Test/Diagnostics/AbstractDiagnosticProviderBasedUserDiagnosticTest.cs

```diff
@@ -75,7 +75,11 @@ public abstract class AbstractDiagnosticProviderBasedUserDiagnosticTest : Abstra
         /// </summary>
         private void AssertNoAnalyzerExceptionDiagnostics(IEnumerable<Diagnostic> diagnostics)
         {
-            Assert.Equal(0, diagnostics.Count(diag => diag.Descriptor.CustomTags.Contains(WellKnownDiagnosticTags.AnalyzerException)));
+            var analyzerExceptionDiagnostics = diagnostics.Where(diag => diag.Descriptor.CustomTags.Contains(WellKnownDiagnosticTags.AnalyzerException));
+
+            // using `Assert.True()` because it allows a custom message to help diagnose the problem
```

Comment:
It sucks that Assert.Empty doesn't just list the elements. Should we just fix that on their end? Or have an AssertEx.Empty until then? I suspect we do this pattern a lot of places.